lib/yolo/models/yolox.ex

```diff
@@ -0,0 +1,208 @@
+defmodule YOLO.Models.Yolox do
+  @moduledoc """
+  YOLOX model implementation for preprocessing input images
+  and postprocessing detections using non-maximum suppression (NMS).
+
+  Supports YOLOX models found at [https://github.com/Megvii-BaseDetection/YOLOX](github.com/Megvii-BaseDetection/YOLOX)
+
+  If using a YOLOX model that was exported with `--decode_in_inference`, you can set
+  `decode_head: false` in the YOLO.detect/3 options.
+  """
+
+  @behaviour YOLO.Model
+
+  import Nx.Defn
+
+  @impl true
+  @spec preprocess(YOLO.Model.t(), term(), Keyword.t()) :: {Nx.Tensor.t(), ScalingConfig}
+  def preprocess(model, image, options) do
+    frame_scaler = Keyword.fetch!(options, :frame_scaler)
+    {_, _channels, height, width} = model.shapes.input
+    {image_nx, image_scaling} = YOLO.FrameScalers.fit(image, {height, width}, frame_scaler)
+
+    image_nx =
+      image_nx
+      |> Nx.as_type({:f, 32})
+      # {h, w, c} -> {c, h, w}
+      |> Nx.transpose(axes: [2, 0, 1])
+      # add another axis {3, 640, 640} -> {1, 3, 640, 640}
+      |> Nx.new_axis(0)
+
+    {image_nx, image_scaling}
+  end
+
+  @impl true
+  def precalculate(_model_ref, shapes, options) do
+    {_, _, height, width} = shapes.input
+
+    {grids, expanded_strides} =
+      generate_grids_and_expanded_strides({width, height}, options[:p6] || false)
+
+    %{grids: grids, expanded_strides: expanded_strides}
+  end
+
+  @doc """
+  Post-processes the model's raw output to produce a filtered list of detected objects.
+
+  Options:
+  * `decode_head` - If true, decode the output head to map predictions to input image space.
+    Defaults to true. Can be set to false if using a YOLOX model that was exported with `--decode_in_inference`
+  * `nms_fun` - Optional custom NMS function. Must calculate detection scores as the product of the maximum class
+    probability and the objectness score.
+  * `prob_threshold` - Minimum probability threshold for detections
+  * `iou_threshold` - IoU threshold for non-maximum suppression
+  """
+  @impl true
+  def postprocess(%{precalculated: precalculated}, model_output, scaling_config, opts) do
+    prob_threshold = Keyword.fetch!(opts, :prob_threshold)
+    iou_threshold = Keyword.fetch!(opts, :iou_threshold)
+    nms_fun = Keyword.get(opts, :nms_fun, &default_nms/3)
+    decode_head? = Keyword.get(opts, :decode_head, true)
+
+    %{grids: grids, expanded_strides: expanded_strides} = precalculated
+
+    model_output
+    |> maybe_decode_head(grids, expanded_strides, decode_head?)
+    |> nms_fun.(prob_threshold, iou_threshold)
+    |> YOLO.FrameScalers.scale_bboxes_to_original(scaling_config)
+  end
+
+  defp maybe_decode_head(model_output, grids, expanded_strides, true = _decode_head?) do
+    process_bboxes(model_output, grids, expanded_strides)
+  end
+
+  defp maybe_decode_head(model_output, _grids, _expanded_strides, false = _decode_head?) do
+    Nx.reshape(model_output, {8400, 85})
```

Comment:
Oh, this is the one thing I need to fix which prevents using this on 416x416 models.

Reply:
(Fixed: https://github.com/poeticoding/yolo_elixir/pull/10/commits/5f5a38098c7a7a94a8d809cf83305e03f493ee0d)